src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/SupplementaryDatasetService.java

```diff
@@ -37,9 +37,9 @@ public void addSupplementaryDatasetEntity(SupplementaryDatasetDTO supplementaryD
           supplementaryDatasetDTO.getPeriodId());
     }
     try {
-      if (existsByExerciseFK(collectionExercise.getExercisePK())) {
+      if (collectionExercise.getSupplementaryDatasetEntity() != null) {
         log.info(
-            "Supplementary dataset with exerciseFk {} has been found.",
+            "Supplementary dataset linked to collection exercise with PK {} found.",
```

Comment:
Is the exercise UUID in scope? This would be preferable in the log entry to the PK.

Reply:
I've updated the log to have the collex uuid instead